streams/readable-byte-streams/read-fully.any.js

```diff
@@ -0,0 +1,539 @@
+
+  {
+    const byobRequest = byobRequests[0];
+    assert_true(byobRequest.nonNull, 'first byobRequest must not be null');
+    assert_true(byobRequest.viewNonNull, 'first byobRequest.view must not be null');
+    const viewInfo = byobRequest.viewInfo;
+    assert_equals(viewInfo.constructor, Uint8Array, 'first view.constructor should be Uint8Array');
+    assert_equals(viewInfo.bufferByteLength, 3, 'first view.buffer.byteLength should be 3');
+    assert_equals(viewInfo.byteOffset, 0, 'first view.byteOffset should be 0');
+    assert_equals(viewInfo.byteLength, 3, 'first view.byteLength should be 3');
+  }
+
+  {
+    const byobRequest = byobRequests[1];
+    assert_true(byobRequest.nonNull, 'second byobRequest must not be null');
+    assert_true(byobRequest.viewNonNull, 'second byobRequest.view must not be null');
+    const viewInfo = byobRequest.viewInfo;
+    assert_equals(viewInfo.constructor, Uint8Array, 'second view.constructor should be Uint8Array');
+    assert_equals(viewInfo.bufferByteLength, 3, 'second view.buffer.byteLength should be 3');
+    assert_equals(viewInfo.byteOffset, 2, 'second view.byteOffset should be 2');
+    assert_equals(viewInfo.byteLength, 1, 'second view.byteLength should be 1');
+  }
+
+}, 'ReadableStream with byte source: readFully(), then multiple enqueue()');
+
+promise_test(async t => {
+  const stream = new ReadableStream({
+    start(c) {
+      const view = new Uint8Array(16);
+      view[0] = 0x01;
+      view[8] = 0x02;
+      c.enqueue(view);
+    },
+    pull: t.unreached_func('pull() should not be called'),
+    type: 'bytes'
+  });
+
+  const byobReader = stream.getReader({ mode: 'byob' });
+  const result1 = await byobReader.readFully(new Uint8Array(8));
+  assert_false(result1.done, 'done');
+
+  const view1 = result1.value;
+  assert_equals(view1.constructor, Uint8Array, 'value.constructor');
+  assert_equals(view1.buffer.byteLength, 8, 'value.buffer.byteLength');
+  assert_equals(view1.byteOffset, 0, 'value.byteOffset');
+  assert_equals(view1.byteLength, 8, 'value.byteLength');
+  assert_equals(view1[0], 0x01);
+
+  byobReader.releaseLock();
+
+  const reader = stream.getReader();
+  const result2 = await reader.read();
+  assert_false(result2.done, 'done');
+
+  const view2 = result2.value;
+  assert_equals(view2.constructor, Uint8Array, 'value.constructor');
+  assert_equals(view2.buffer.byteLength, 16, 'value.buffer.byteLength');
+  assert_equals(view2.byteOffset, 8, 'value.byteOffset');
+  assert_equals(view2.byteLength, 8, 'value.byteLength');
+  assert_equals(view2[0], 0x02);
+}, 'ReadableStream with byte source: enqueue(), readFully() partially, then read()');
+
+promise_test(async () => {
+  let controller;
+
+  let pullCount = 0;
+  const byobRequestDefined = [];
+  let byobRequestViewDefined;
+
+  const stream = new ReadableStream({
+    start(c) {
+      controller = c;
+    },
+    async pull() {
+      byobRequestDefined.push(controller.byobRequest !== null);
```

Comment:
I'd mildly prefer to make `pull` a `step_func` and assert these things here rather than below, but whichever is fine.

Reply:
...Does `step_func` work correctly with `async` functions? I don't think it will await the returned promise to handle any rejections with an `AssertionError`. 😕

Reply:
Ah, I see the problem. Yeah, that won't work. Never mind.